src-ghc/Pact/GasModel/GasTests.hs

```diff
@@ -17,14 +17,12 @@ import Data.Bool (bool)
 import Data.Default (def)
 import Data.List (foldl')
 import Data.Maybe (fromMaybe)
-import Data.List.NonEmpty (NonEmpty(..))
```

Comment:
I don't disagree with this change, but it's a lot of modifications in a module that is soon going to disappear, once the next generation of gas tests are complete. In particular this file will be replaced by something more general, based on hedgehog, that can both compute the nominal gas cost for individual instructions — as this file does — but also exercise nested use of instructions to find more edge cases. The rest of the GasModel code will still be used, but not this module.

Reply:
See other comment. This code was extremely punishing to deal with/add tests to. If the new gas tests come soon great but in the meantime these changes radically improve the ability for maintainers to add new cases.

Reply:
Understood.

Reply:
I don't mean to dismiss the concerns raised. We can consider the following:

1. I can easily move this to a new PR. However given the short lifespan of the GasTests it may not be worth that much attention.

2. I can leave it alone and simply add the missing gas tests, since as you note it's a large diff on a module that's going away.

Reply:
That's very considerate, @sirlensalot! I'd prefer, now that we've come this far, to see these changes go through in this PR. I'll then update my new tests to drop NonEmpty. The code I'm working on still uses several of the previous GasModel modules, which you've also touched, so not all of this work will be lost, just mainly the parts in `GasTests.hs`.